config/jobs/kubernetes/sig-node/Makefile

```diff
@@ -0,0 +1,17 @@
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+.PHONY: generate
+generate:
```

Comment:
We should have a step to verify that contributors are not modified the generated yaml without running this.

ie I'd like a job to make sure that the generation code matches what is the repo so we can confirm that people didn't edit this manually.

Reply:
Some kops jobs in test-infra are already using something like this. We should look if they have verification and do the same, or add it for both.

Reply:
I'm going to remove Makefile and gen.py from this PR in order to test dra-canary.yaml without modifying anything else. They're here just for you to understand and review the approach.

If canary jobs work, I'll submit Makefile, gen.py and some kind of verification approach as a separate PR.

Would it work for you this way?

Reply:
I prefer to use this PR to finalize the complete solution for generating jobs.

I don't see value in merging canary jobs just to test that they work. A diff against the non-canary variant should give us enough confidence that they will, because the diff should be minimal.

Reply:
I don't fully agree as merging only dra-canary.yaml wouldn't break anything. Merging dra-pull.yaml and dra-ci.yaml can break existing jobs, at least potentially. Why do we want to risk?

Reply:
What's the benefit of merging only `dra-canary.yaml`?

Reply:
Thinking aloud: we could merge them, run them, then overwrite the normal jobs with something that closely mirrors those canaries. Okay, that makes sense to me.

As we have the full discussion around generation here, perhaps create a separate PR with the dra-canary.yaml and then finish this PR?

Reply:
But let's do that for the "final" `dra-canary.yaml` - I haven't look at the current state again and won't get to it for a few more days (vacation).

Reply:
Sure, I'll create a new pr after we finalise dra-canary.yaml generation and verification. I'll post a final diff when I'm done.